tracks/csharp/exercises/nucleotide-count/mentoring.md

```diff
@@ -0,0 +1,57 @@
+### Reasonable solutions
+
+### Using a foreach loop
+
+```csharp
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+public static class NucleotideCount
+{
+    public static IDictionary<char, int> Count(string sequence)
+    {
+        var result = new Dictionary<char, int> { {'A', 0}, {'C', 0}, {'G', 0}, {'T', 0} };
+        
+        foreach (var c in sequence)
+        {
+            if (result.ContainsKey(c)) result[c]++;
+            else throw new ArgumentException("Invalid Nucleotide Symbol");
+        }
+
+        return result;
+    }
+}
+```
+
+### Foreach with ToDictionary
+
+```csharp
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+public static class NucleotideCount
+{
+    private static readonly char[] ValidNucleotideSymbols = {'A', 'C', 'G', 'T'};
+    
+    public static IDictionary<char, int> Count(string sequence)
+    {
+        var result = ValidNucleotideSymbols.ToDictionary(c => c, c => 0);
```

Comment:
```suggestion
        var nucleotideCounts = ValidNucleotideSymbols.ToDictionary(c => c, c => 0);
```

Maybe use a more descriptive name?

Reply:
Good point, is more descriptive

Reply:
Resolved in 9fd8d58832c4d19e959cdf60c4d2429ec951bf12